compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1604,25 +1608,34 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
             })
         }
 
+        let real_rust_source_base_dir = sess.real_rust_source_base_dir();
```

Comment:
This is the only place where `real_rust_source_base_dir` gets called. Should I just move that logic into this file?

Reply:
Also, should I be worried about perf here? We now re-compute `real_rust_source_base_dir` each time `imported_source_file` gets called. Not sure what would be the right place for a cache though.

Reply:
Before https://github.com/rust-lang/rust/pull/84233 this got computed *after* the `config` callback, in `build_session`. That would be a lot better. For it seems that was changed for good reasons.